internal/server/iigointernal/monitoring.go

```diff
@@ -27,19 +29,34 @@ func (m *monitor) addToCache(roleToMonitorID shared.ClientID, variables []rules.
 	}
 }
 
-func (m *monitor) monitorRole(roleAccountable baseclient.Client) shared.MonitorResult {
+func (m *monitor) monitorRole(g *gamestate.GameState, roleAccountable baseclient.Client) shared.MonitorResult {
 	roleToMonitor, roleName, err := m.findRoleToMonitor(roleAccountable.GetID())
 	if err == nil {
 		decideToMonitor := roleAccountable.MonitorIIGORole(roleName)
 		evaluationResult := false
 		if decideToMonitor {
 			evaluationResult = m.evaluateCache(roleToMonitor, rules.RulesInPlay)
-			evaluationResult, announce := roleAccountable.DecideIIGOMonitoringAnnouncement(evaluationResult)
-			if announce {
-				message := generateMonitoringMessage(roleName, evaluationResult)
-				broadcastToAllIslands(roleAccountable.GetID(), message)
-			}
 		}
+
+		evaluationResultAnnounce, announce := roleAccountable.DecideIIGOMonitoringAnnouncement(evaluationResult)
+
+		//announce == decideToMonitor
+		variablesToCache := []rules.VariableFieldName{rules.MonitorRoleAnnounce, rules.MonitorRoleDecideToMonitor}
+		valuesToCache := [][]float64{{boolToFloat(decideToMonitor)}, {boolToFloat(announce)}}
+		m.addToCache(roleAccountable.GetID(), variablesToCache, valuesToCache)
+
+		if announce {
+			//check if evalResult = o.g. evalResult
+			variablesToCache := []rules.VariableFieldName{rules.MonitorRoleEvalResult, rules.MonitorRoleEvalResultDecide}
+			valuesToCache := [][]float64{{boolToFloat(evaluationResult)}, {boolToFloat(evaluationResultAnnounce)}}
+			m.addToCache(roleAccountable.GetID(), variablesToCache, valuesToCache)
+
+			message := generateMonitoringMessage(roleName, evaluationResult)
+			broadcastToAllIslands(roleAccountable.GetID(), message)
+
+			g.IIGOTurnsInPower[roleName] = m.TermLengths[roleName] + 1
```

Comment:
The roles seem to be incremented every round, why are they again incremented here?

Reply:
This is not incrementing, this is setting it to be just over the term length. Roles have an obligation to hold an election if the role to be elected has gone over term. In monitoring when a role decides to announce that a cheater role has broken the rules, this makes it so that it is now the other role, who has the power to hold elections for the cheater role, is obliged to hold that election.